### ISSUE TITLE
Restructure the current API Fetch and Display Process

### ISSUE THREAD

**Issue by rumaan** [rumaan/nasa_apod_flutter#7]
- The current implementation doesn't exactly follow BLoC principles
- Suggestion to refactor and restructure without breaking anything

**Comment by Lukitaduarte** [comment]
what do you think about use https://bloclibrary.dev/ ?

**Comment by rumaan** [comment]
> what do you think about use https://bloclibrary.dev/ ?

Yes, I'd really appreciate it if someone could help me out with setting that app with Bloc and refactor the project accordingly!
